tools/velvet/velveth.xml

```diff
@@ -1,70 +1,97 @@
 <tool id="velveth" name="velveth" version="@WRAPPER_VERSION@.0">
   <description>Prepare a dataset for the Velvet velvetg Assembler</description>
-  <version_command>velveth 2&gt;&amp;1 | grep "Version" | sed -e 's/Version //'</version_command>
   <macros>
     <import>macros.xml</import>
   </macros>
   <expand macro="requirements"/>
   <expand macro="stdio"/>
-  <command interpreter="python">
-    velveth_wrapper.py
-           '$out_file1' '$out_file1.extra_files_path'
-           $hash_length
-           $strand_specific
-           #for $i in $inputs
-                ${i.file_format}
-                ${i.read_type}
-                ${i.input}
-           #end for
-  </command>
+  <version_command>velveth 2&gt;&amp;1 | grep "Version" | sed -e 's/Version //'</version_command>
+  <command><![CDATA[
+    export OMP_NUM_THREADS="\${GALAXY_SLOTS:-1}" &&
+    export OMP_THREAD_LIMIT="\${GALAXY_SLOTS:-1}" &&
+    velveth
+      '$out_file1.extra_files_path'
+      $hash_length
+      #for $i in $inputs
+          ${i.file_format}
+          ${i.read_type}
+          #if str( $i.input_type.input_type_selector ) == "paireds":
+              -separate $i.input1 $i.input2
+          #else if str( $i.input_type.input_type_selector ) == "pairedi":
+              -interleaved ${i.input}
+          #else:
+              ${i.input}
+          #end if
+      #end for
+      $strand_specific
+  ]]></command>
   <inputs>
-    <param label="Hash Length" name="hash_length" type="select" help="k-mer length in base pairs of the words being hashed.">
-      <option value="11">11</option>
-      <option value="13">13</option>
-      <option value="15">15</option>
-      <option value="17">17</option>
-      <option value="19">19</option>
-      <option value="21" selected="yes">21</option>
-      <option value="23">23</option>
-      <option value="25">25</option>
-      <option value="27">27</option>
-      <option value="29">29</option>
-    </param>
+      <param label="Hash Length" name="hash_length" type="integer" value="21" min="11" max="191" help="k-mer length in base pairs of the words being hashed."/>
```

Comment:
is it still true that we need to compile in the max kmer length?

Reply:
seems so. at least velvet outputs this on startup. was wondering if we can parse this and use the value in the form. is there such a possibility?

Reply:
I always just compile velvet with Max kmer size 191, omp turned on and categories 4 as this cover most things... No point using the defaults, they're useless.. they were conceived a long time ago